File: example/src/main/java/ru/virgil/example/user/UserDetailsRepository.java
package ru.virgil.example.user;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import ru.virgil.example.security.SecurityUserDetails;

import java.util.Optional;
import java.util.UUID;

@Repository
public interface UserDetailsRepository extends JpaRepository<UserDetails, UUID> {

    Optional<UserDetails> findBySecurityUserDetails(SecurityUserDetails securityUserDetails);

}

File: example/src/main/java/ru/virgil/example/user/UserDetailsRepository.kt
package ru.virgil.example.user

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import ru.virgil.example.security.SecurityUserDetails
import java.util.*

@Repository
interface UserDetailsRepository : JpaRepository<UserDetails, UUID> {

    fun findBySecurityUserDetails(securityUserDetails: SecurityUserDetails): Optional<UserDetails>
}
